statsd_client.ml

```diff
@@ -153,3 +152,47 @@ module Lwt = Make (
     let socket dom typ proto = Lwt_unix.socket dom typ proto
   end
 )
+
+module Async = Make (
+  struct
+    open Core
+    open Async
+
+    let ipaddr () = !ipaddr
+    let port () = !port
+
+    type 'a _r = 'a Deferred.t
+    let ( >>= ) = ( >>= )
+    let return = return
+    let catch f error =
+      try_with f >>= function
+      | Ok d -> return d
+      | Error e -> error e
+    let list_iter f l = Deferred.List.iter ~f l
+    let close fd = Unix.close fd
+
+    type file_descr = Async_unix.Fd.t
+
+    let socket domain typ protocol =
+      let socket_type =
+        match domain, typ, protocol with
+        | Unix.PF_INET, Unix.SOCK_DGRAM, 17 -> Socket.Type.udp
+        | Unix.PF_INET, Unix.SOCK_STREAM, 6 -> Socket.Type.tcp
```

Comment:
What are 17 and 6?

Reply:
IP protocol numbers. https://en.wikipedia.org/wiki/List_of_IP_protocol_numbers
Put a comment and renamed `protocol` to `ip_protocol_num`